filterable-file-log/serializer.lua

```diff
@@ -60,11 +64,31 @@ function _M.serialize(ngx, filters)
     service = ngx.ctx.service,
     api = ngx.ctx.api,
     consumer = ngx.ctx.authenticated_consumer,
+    jwt_claims = jwt_claims,
     client_ip = ngx.var.remote_addr,
     started_at = ngx.req.start_time() * 1000
   }
 end
 
+function get_jwt_claims(headers)
+  if not headers['authorization'] then
+    return nil
+  end
+
+  local token = string.match(headers['authorization'], 'Bearer (.+)')
+  if not token then
+    return nil
+  end
+
+  local encoded_claims = stringx.split(token, '.')[2]
+  if not encoded_claims then
+    return nil
+  end
+
+  local claims = ngx.decode_base64(encoded_claims)
```

Comment:
🤔 It is, but I'm not sure why. It might just be global? Or maybe it's a reference to the local ngx that's passed into `_M.serialize` since we're only calling `get_jwt_claims` from within `_M.serialize`? I'll do some more investigating

Reply:
>The API is exposed to Lua in the form of two standard packages ngx and ndk. These packages are in the default global scope within ngx_lua and are always available within ngx_lua directives.
https://openresty-reference.readthedocs.io/en/latest/Lua_Nginx_API/

So `ngx` is a global var – presumably there's no reason for it to be passed into `_M.serialize`.

Reply:
https://github.com/Kong/kong/commit/c747fb487326fb0b2893fbb95a2a7c6035e6a984#diff-ffbe4a2029a35f89cb0c13a63b8b5d35

This commit is similarly confusing. There are many things about lua scope that I don't understand!